static/js/message_edit.js

```diff
@@ -181,6 +181,17 @@ export function get_deletability(message) {
     return false;
 }
 
+export function can_move_message(message) {
+    if (!page_params.realm_allow_message_editing) {
+        return false;
+    }
+
+    const is_stream_editable =
+        message.is_stream && settings_data.user_can_move_messages_between_streams();
+
+    return get_editability(message) !== editability_types.NO || is_stream_editable;
```

Comment:
Does this correctly calculate whether you can edit a topic? I seem to recall some special behavior for "no topic", for example, that is not reflected here.

Reply:
I think we can show the menu option and have the hotkey work whenever you're allowed to move a message, regardless of whether that's because the topic is "no topic" or not.

Reply:
Yeah that sounds right to me.